ngraph/core/include/ngraph/op/einsum.hpp

```diff
@@ -0,0 +1,62 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "ngraph/node.hpp"
+#include "ngraph/op/op.hpp"
+
+namespace ngraph
+{
+    namespace op
+    {
+        namespace v7
+        {
+            /// \brief Einsum operation.
+            class NGRAPH_API Einsum : public Op
+            {
+            public:
+                NGRAPH_RTTI_DECLARATION;
+
+                Einsum() = default;
+
+                ///
+                /// \brief      Constructs Einsum operation.
+                ///
+                /// \param      inputs        Input nodes on which Einsum operation performs
+                /// contraction
+                ///
+                /// \param      equation      Einstein summation convention
+                ///
+                Einsum(const OutputVector& inputs, const std::string& equation);
+
+                void validate_and_infer_types() override;
+
+                bool visit_attributes(AttributeVisitor& visitor) override;
+
+                std::shared_ptr<Node>
+                    clone_with_new_inputs(const OutputVector& new_args) const override;
+
+            private:
+                std::string m_equation;
+
+                /// \brief      Check equation format and extracts input subscripts and output
+                /// subscript
+                ///
+                /// \param      input_subscripts      A vector of extracted input subscripts
+                /// \param      output_subscript      An output subscript
+                ///
```

Comment:
These two methods below looks really helpful, so I would consider making them public.

Reply:
@lazarevevgeny Why do we need to make these methods public? Would users use their?

Reply:
@lazarevevgeny, @ilyachur, I propose to make these methods static if we want them to be public.

Reply:
Let us left them as `private` methods because now I see no other use cases, as @ilyachur noticed.

Reply:
Consider the plugin owner who wants to implement Einsum operation. In case when the equation is implicit then they probably need to calculate the output subscript themselves. So each and every plugin should do this. If we provide them a method to do this this will be super helpful.

Reply:
The same story for the labels extracting. It will be very convenient for them to have such a method to get the order of subscripts.

Reply:
Done.